library/core/class.format.php

```diff
@@ -2464,6 +2464,56 @@ public static function rich(string $deltas): string {
         return $renderer->render($blotGroups);
     }
 
+    /**
+     * Generate a quote to embed in a Rich quote for all existing formats.
+     *
+     * @param string|array $body The string or array body content of the post.
+     * @param string $format The initial format of the post.
+     * @return string
+     * @throws \Garden\Container\ContainerException
+     * @throws \Garden\Container\NotFoundException
+     */
+    public static function quoteEmbed($body, string $format): string {
+        if ($format === "Rich") {
+            if (is_string($body)) {
+                $body = json_decode($body, true);
+            }
+            return self::richQuote($body);
+        } else {
+            $previousLinksValue = c('Garden.Format.Links');
+            saveToConfig('Garden.Format.Links', false, ['Save' => false]);
+            $value = self::to($body, $format);
+
+            // These breaks make the collapsing behaviour much more difficult in a rich quote.
+            // Replace them with starting and closing p tags.
+            $value = str_replace("<br>", "</p><p>", $value);
```

Comment:
This seems a little dangerous. Are you sure the break tag is only ever going to be the direct child of paragraph tags?